controller/array_action/array_mediator_ds8k.py

```diff
@@ -617,32 +618,47 @@ def get_iscsi_targets_by_iqn(self, host_name):
 
     def get_array_fc_wwns(self, host_name):
         logger.debug("getting the connected fc port wwpns for host {} from array".format(host_name))
-
+        api_host = self._get_api_host_by_name(host_name)
+        if api_host is None:
+            return []
+        wwpns = [port[LOGIN_PORT_WWPN] for port in api_host.login_ports if
+                 port[LOGIN_PORT_STATE] == LOGIN_PORT_STATE_ONLINE]
+        logger.debug("found wwpns: {}".format(wwpns))
+        return wwpns
+
+    def _get_api_host_by_name(self, host_name):
         try:
-            host = self.client.get_host(host_name)
-            wwpns = [port[LOGIN_PORT_WWPN] for port in host.login_ports if
-                     port[LOGIN_PORT_STATE] == LOGIN_PORT_STATE_ONLINE]
-            logger.debug("found wwpns: {}".format(wwpns))
-            return wwpns
+            return self.client.get_host(host_name)
         except exceptions.NotFound:
-            raise array_errors.HostNotFoundError(host_name)
+            return None
         except exceptions.ClientException as ex:
-            logger.error(
-                "failed to get array fc wwpn. Reason is: {}".format(ex.details)
-            )
             raise ex
 
+    def _get_fc_wwns_from_api_host(self, api_host):
+        host_ports = api_host.host_ports_briefs
+        return [p["wwpn"] for p in host_ports]
+
+    def get_host_by_name(self, host_name):
+        api_host = self._get_api_host_by_name(host_name)
+        if api_host is None:
+            return None
+        fc_wwns = self._get_fc_wwns_from_api_host(api_host)
+        connectivity_types = []
+        if fc_wwns:
+            connectivity_types.append(config.FC_CONNECTIVITY_TYPE)
+        initiators = Initiators(nvme_nqns=[], fc_wwns=fc_wwns, iscsi_iqns=[])
```

Comment:
could we make all the parameters optional?
so this could simply be:
```suggestion
        initiators = Initiators(fc_wwns=fc_wwns)
```

Reply:
done